magicctapipe/scripts/lst1_magic/semi_automatic_scripts/merging_runs.py

```diff
@@ -142,165 +145,81 @@ def merge(target_dir, identification, MAGIC_runs, env_name, source, NSB_match):
-                            )
-            elif identification == "1_M1M2":
-                if os.path.exists(f"{MAGIC_DL1_dir}/M1") & os.path.exists(
-                    f"{MAGIC_DL1_dir}/M2"
-                ):
-                    dates = [
-                        os.path.basename(x) for x in glob.glob(f"{MAGIC_DL1_dir}/M1/*")
-                    ]
-                    for i in dates:
-                        runs = [
-                            os.path.basename(x)
-                            for x in glob.glob(f"{MAGIC_DL1_dir}/M2/{i}/*")
-                        ]
-                        for r in runs:
-                            if (len(glob.glob(f"{MAGIC_DL1_dir}/M1/{i}/{r}")) > 0) and (
-                                len(glob.glob(f"{MAGIC_DL1_dir}/M2/{i}/{r}")) > 0
-                            ):
-                                os.makedirs(f"{MAGIC_DL1_dir}/Merged/{i}/Merged/logs")
-                                f.write(
-                                    f"conda run -n {env_name} merge_hdf_files --input-dir {MAGIC_DL1_dir}/Merged/{i}/{r} --output-dir {MAGIC_DL1_dir}/Merged/{i}/Merged --run-wise >{MAGIC_DL1_dir}/Merged/{i}/Merged/logs/merge_{i}_{r}_"
-                                    + "${SLURM_JOB_ID}.log \n"
-                                )
-            else:
-                dates = [
-                    os.path.basename(x) for x in glob.glob(f"{MAGIC_DL1_dir}/M1/*")
-                ]
-                for i in dates:
-
-                    if not os.path.exists(f"{MAGIC_DL1_dir}/Merged/{i}/Merged"):
-
-                        continue
-
-                    if (
-                        len(glob.glob(f"{MAGIC_DL1_dir}/Merged/{i}/Merged/*MAGIC*.h5"))
-                        > 0
-                    ):
-
-                        os.makedirs(f"{MAGIC_DL1_dir}/Merged/Merged_{i}/logs")
-                        f.write(
-                            f"conda run -n {env_name} merge_hdf_files --input-dir {MAGIC_DL1_dir}/Merged/{i}/Merged --output-dir {MAGIC_DL1_dir}/Merged/Merged_{i} >{MAGIC_DL1_dir}/Merged/Merged_{i}/logs/merge_night_{i}_"
-                            + "${SLURM_JOB_ID}.log \n"
-                        )
+                    rc = rc_lines(
+                        store=f"{indir} ${{SLURM_JOB_ID}}", out=f"{outdir}/logs/list"
+                    )
+                    f.writelines(rc)
+                    os.system(f"echo {indir} >> {outdir}/logs/list_dl0.txt")
+                else:
+                    print(
+                        f"ERROR {MAGIC_DL1_dir}/M1/{i[0]}/{i[1]} or {MAGIC_DL1_dir}/M2/{i[0]}/{i[1]} does not exist"
+                    )
+        else:
+            dates = np.unique(MAGIC_runs.T[0])
+            for i in dates:
+                if not os.path.exists(f"{MAGIC_DL1_dir}/Merged/{i}/Merged"):
+                    continue
+
+                indir = f"{MAGIC_DL1_dir}/Merged/{i}/Merged"
+                outdir = f"{MAGIC_DL1_dir}/Merged/Merged_{i}"
```

Comment:
because we first merge M1/2 subruns to runs (in Merged/{i}), then M1 with M2 run-wise (in Merged/{i}/Merged) then runs to nights (in Merged/Merged_{i}), {i}=night

Reply:
I think we are overdoing it, probably any one of those 3 would suffice for running the coincidence script, but I kept the original code for now